core/mutex.c

```diff
@@ -54,12 +55,11 @@ void mutex_lock(struct mutex_t *mutex)
 static void mutex_wait(struct mutex_t *mutex)
 {
     unsigned irqstate = disableIRQ();
-    DEBUG("%s: Mutex in use. %u\n", sched_active_thread->name, mutex->val);
+    DEBUG("%s: Mutex in use. %u\n", sched_active_thread->name, ATOMIC_VALUE(mutex->val));
 
-    if (mutex->val == 0) {
+    if (atomic_set_to_one(&mutex->val)) {
```

Comment:
in the current version it gets changed if it is set to zero, see L61 in the original file. In the PR version it gets changed if the value is set to zero.